src/fiatExchanges/FiatExchangeCurrency.tsx

```diff
@@ -0,0 +1,250 @@
+import BackButton from 'src/components/BackButton'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import i18n from 'src/i18n'
+import RadioButton from 'src/icons/RadioButton'
+import { emptyHeader } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import variables from 'src/styles/variables'
+import { Currency } from 'src/utils/currencies'
+import { CICOFlow, FiatExchangeFlow } from './utils'
+
+type RouteProps = StackScreenProps<StackParamList, Screens.FiatExchangeCurrency>
+type Props = RouteProps
+
+export const fiatExchangesOptionsScreenOptions = ({
+  route,
+}: {
+  route: RouteProp<StackParamList, Screens.FiatExchangeCurrency>
+}) => {
+  const { flow } = route.params
+  return {
+    ...emptyHeader,
+    headerLeft: () => (
+      <BackButton
+        eventName={FiatExchangeEvents.cico_select_currency_back}
+        eventProperties={{ flow }}
+      />
+    ),
+    headerTitle: i18n.t(`fiatExchangeFlow.${route.params.flow}.selectCurrencyTitle`),
+    headerRightContainerStyle: { paddingRight: 16 },
+  }
+}
+
+function CurrencyRadioItem({
+  selected,
+  onSelect,
+  enabled = true,
+  title,
+  body,
+  containerStyle,
+  testID,
+}: {
+  selected: boolean
+  onSelect: () => void
+  enabled?: boolean
+  title: string
+  body?: string
+  containerStyle: ViewStyle
+  testID?: string
+}) {
+  return (
+    <TouchableWithoutFeedback onPress={onSelect} disabled={!enabled}>
+      <View
+        style={[
+          styles.currencyItemContainer,
+          containerStyle,
+          { borderColor: selected ? colors.greenUI : colors.gray3 },
+        ]}
+        testID={testID}
+      >
+        <RadioButton selected={selected} disabled={!enabled} />
+        <Text style={[styles.currencyItemTitle, enabled ? {} : { color: colors.gray3 }]}>
+          {title}
+        </Text>
+        {body && <Text style={styles.currencyItemBody}>{body}</Text>}
+      </View>
+    </TouchableWithoutFeedback>
+  )
+}
+
+function FiatExchangeCurrency({ route, navigation }: Props) {
+  const { t } = useTranslation()
+  const { flow } = route.params
+
+  const [selectedCurrency, setSelectedCurrency] = useState<Currency>(Currency.Dollar)
+
+  const goToProvider = () => {
+    ValoraAnalytics.track(FiatExchangeEvents.cico_option_chosen, {
+      flow,
+      currency: selectedCurrency,
+    })
+    if (flow === FiatExchangeFlow.Spend) {
+      return navigate(Screens.BidaliScreen, {
+        currency: selectedCurrency,
+      })
+    }
+    navigate(Screens.FiatExchangeAmount, {
+      currency: selectedCurrency,
+      flow: flow === FiatExchangeFlow.CashIn ? CICOFlow.CashIn : CICOFlow.CashOut,
+    })
+  }
+
+  const onSelectCurrency = (currency: Currency) => () => setSelectedCurrency(currency)
```

Comment:
I'm not sure what does this wrapper function do, can just directly use `setSelectedCurrency(currency)` ?